cognite/neat/rules/exporter/_rules2dms.py

```diff
@@ -521,56 +498,186 @@ def find_existing_views(self, client: CogniteClient) -> set:
+                )
+                logging.info(f"View {view_id}/{self.views[view_id].version} already exists! Skipping !!!")
+
+    def create_data_model(
+        self,
+        client: CogniteClient,
+        existing_data_model: DataModelId | None = None,
+    ):
+        if not existing_data_model:
+            logging.info(f"Creating data model {self.external_id} version {self.version} in space {self.space}")
+            _ = client.data_modeling.data_models.apply(
+                DataModelApply(
+                    name=self.name,
+                    description=self.description,
+                    space=self.space,
+                    external_id=self.external_id,
+                    version=self.version,
+                    views=list(self.views.values()),
+                )
             )
-        )
+        else:
+            logging.info(f"Data model {self.space}:{self.external_id}/{self.version} already exists! Skipping !!!")
 
-    def remove_data_model(self, client: CogniteClient):
-        """Helper function to remove a data model, and all underlying views and containers from CDF.
+    def remove(self, client: CogniteClient, components_to_remove: set | None = None):
+        """Remove DMS schema components from CDF.
 
         Args:
-            client: Cognite client.
+            client: Connected Cognite client.
+            components_to_remove: Which components to remove. Takes set
+
+        !!! note "Component Creation Policy"
+            Here is more information about the different component creation policies:
+            - `all`: all components of the data model will be created, meaning space, containers, views and data model
+            - `data model`: only the data model will be created
+            - `view`: only the views will be created
+            - `container`: only the containers will be created
         """
 
+        if components_to_remove is None:
+            components_to_remove = set(["all"])
+        if "data model" in components_to_remove or "all" in components_to_remove:
+            self.remove_data_model(client)
+        if "view" in components_to_remove or "all" in components_to_remove:
+            self.remove_views(client)
+        if "container" in components_to_remove or "all" in components_to_remove:
+            self.remove_containers(client)
+        if "space" in components_to_remove or "all" in components_to_remove:
+            self.remove_spaces(client)
+
+    def remove_data_model(self, client: CogniteClient):
         if client.data_modeling.data_models.retrieve((self.space, self.external_id, self.version)):
-            logging.info(f"Removing data model {self.external_id} version {self.version} from space {self.space}")
+            logging.info(f"Removing data model {self.space}:{self.external_id}/{self.version}")
             _ = client.data_modeling.data_models.delete((self.space, self.external_id, self.version))
 
+    def remove_views(self, client: CogniteClient):
```

Comment:
For all of these function, avoid removing one view, container and space at at time. Rewrite to one request

Reply:
All at one implemented